Source/stash.h

```diff
@@ -0,0 +1,57 @@
+/**
+ * @file stash.h
+ *
+ * Interface of player stash.
+ */
+#pragma once
+
+#include <cstdint>
+
+#include "engine/point.hpp"
+#include "items.h"
+#include "loadsave.h"
+#include "palette.h"
+#include "player.h"
+#include <map>
+
+namespace devilution {
+
+/**
+ * Identifiers for each of the stash squares
+ * @see #StashRect
+ */
+enum StashSlot : uint8_t {
+	// clang-format off
+	STASH_SLOT_FIRST = 0,
+	STASH_SLOT_LAST  = 99,
+	STASH_SLOT_NUM = 100
+	// clang-format on
+};
+
+struct StashStruct {
+	void RemoveStashItem(int8_t iv);
+	std::map<int, std::array<int8_t, STASH_SLOT_NUM>> stashGrids;

```

Comment:
```suggestion
	std::map<int, std::array<std::ptrdiff_t, STASH_SLOT_NUM>> stashGrids;
```
or std::size_t if the logic is refactored to not rely on negative values would allow more items to be stored.

Reply:
I have gone for `uint16_t` to be sure that it matches with the save format.